reconcile/utils/aws_api.py

```diff
@@ -889,7 +889,7 @@ def get_amis_details(self,
                          account: dict[str, Any],
                          owner_account: dict[str, Any],
                          regex: str,
-                         region: Optional[str] = None) -> List[Dict[str, Sequence[Collection[str]]]]:
+                         region: Optional[str] = None) -> List[Dict[str, Any]]:
```

Comment:
Any is a valid choice here in my opinion. the `item` constructed below has mixed type contents so automatic inference is hard. you could theoretically introduce a dataclass to represent the item but that would be a bit overengineered.